cosmos.go

```diff
@@ -356,7 +373,7 @@ func (c *cosmosImpl) executeAsync(query string, asyncResponses *[]interfaces.Asy
 	if err := c.pool.ExecuteAsync(query, intermediateChannel); err != nil {
 		return nil, err
 	}
-	errorCallback(err)
+	defer errorCallback(err)
```

Comment:
Should not be deferred, as this should return the error of the first call to executeAsync otherwise it is not async anymore.
```suggestion
	errorCallback(err)
```

Reply:
done